cmd/zoekt-webserver/main.go

```diff
@@ -504,3 +510,70 @@ var (
 		Help: "The total number of errors from zoekt watchdog.",
 	})
 )
+
+func mustRegisterMemoryMapMetrics(l sglog.Logger) {
+	if runtime.GOOS != "linux" {
+		l.Debug(
+			"skipping registration: must be running on a linux-based operating system",
+			sglog.String("current_operating_system", runtime.GOOS),
+			sglog.String("desired_operating_system", "linux"))
+
+		return
+	}
+
+	fs, err := procfs.NewDefaultFS()
+	if err != nil {
+		panic(fmt.Sprintf("registering memory map metrics: failed to initialize proc FS: %s", err))
+	}
+
+	prometheus.MustRegister(prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Name: "zoekt_webserver_max_memory_map_count",
```

Comment:
minor: maybe consider a more generic name for this. Then we can look into generalising this code and exporting more useful stats in all of our services. Prometheus will already attach labels to this which indicate it is for webserver.